### PR TITLE
add more color stuff for vo_dmabuf_wayland

### PR DESCRIPTION
**Pull request by Dudemanguy**
~1. Support `--target-prim` and `--target-trc`.~
2. Map PL_COLOR_TRC_BT_1886 to XX_COLOR_MANAGER_V4_TRANSFER_FUNCTION_BT709

### CODE REVIEW

File: video/out/vo_dmabuf_wayland.c [hunk 1]
```diff
@@ -700,16 +701,26 @@ static int reconfig(struct vo *vo, struct mp_image *img)
...
     mp_image_params_restore_dovi_mapping(&p->target_params);
     // Strip metadata that is not understood anyway.
     struct pl_hdr_metadata *hdr = &p->target_params.color.hdr;
     hdr->scene_max[0] = hdr->scene_max[1] = hdr->scene_max[2] = 0;
     hdr->scene_avg = hdr->max_pq_y = hdr->avg_pq_y = 0;
+
+    if (gl_opts->target_prim)
+        p->target_params.color.primaries = gl_opts->target_prim;
+    if (gl_opts->target_trc)
+        p->target_params.color.transfer = gl_opts->target_trc;
```
Comment (kasper93) [inline video/out/vo_dmabuf_wayland.c:719]: Why not use `-vf=format` to adjust the video/output? `--target-*` options are meant to be used to convert video to given format.

Comment (Dudemanguy) [reply video/out/vo_dmabuf_wayland.c:719]: Well the idea here is that the compositor would be the one doing the converting.

Comment (kasper93) [reply video/out/vo_dmabuf_wayland.c:719]: but with this protocol we say what format is the video, not really what is outputted from compositor. Also `vf=format` by default doesn't do any conversions and just overrides video tagged colorspace, so it is perfect for the job here.

Comment (Dudemanguy) [reply video/out/vo_dmabuf_wayland.c:719]: Yeah you're right. Will drop this.

**Review by kasper93 — APPROVED** [review]

**Pull request merged**